src/test/java/io/openliberty/tools/intellij/it/GradleSingleModMPSIDProjectTest.java

```diff
@@ -0,0 +1,258 @@
+     */
+    private final String SM_MP_PROJECT_RES_URI = "api/resource";
+
+    /**
+     * Project response.
+     */
+    private final String SM_MP_PROJECT_OUTPUT = "Hello! Welcome to Open Liberty";
+
+    /**
+     * Relative location of the WLP installation.
+     */
+    private final String WLP_INSTALL_PATH = "build";
+
+    /**
+     * The path to the test report.
+     */
+    private final Path TEST_REPORT_PATH = Paths.get(PROJECTS_PATH_NEW, SM_MP_PROJECT_NAME_NEW, "build", "reports", "tests", "test", "index.html");
+
+    /**
+     * Build file name.
+     */
+    private final String BUILD_FILE_NAME = "build.gradle";
+
+    /**
+     * Action command to open the build file.
+     */
+    private final String BUILD_FILE_OPEN_CMD = "Liberty: View Gradle config";
+
+    /**
+     * Dev mode configuration start parameters.
+     */
+    private final String DEV_MODE_START_PARAMS = "--hotTests";
+
+    /**
+     * Dev mode configuration custom start parameters for debugging.
+     */
+    private final String DEV_MODE_START_PARAMS_DEBUG = "--libertyDebugPort=9876";
+
+    /**
+     * Prepares the environment for test execution.
+     */
+    @BeforeAll
+    public static void setup() {
+        try {
+            // Copy the directory from PROJECTS_PATH to PROJECTS_PATH_NEW
+            TestUtils.copyDirectory(PROJECTS_PATH, PROJECTS_PATH_NEW);
+
+            Path pathNew = Path.of(PROJECTS_PATH_NEW);
+            Path projectDirPath = pathNew.resolve(SM_MP_PROJECT_NAME);
+
+            // Define paths for the original and copy of settings.gradle
+            Path originalPath = projectDirPath.resolve("settings.gradle");
+            Path originalPathCopy = projectDirPath.resolve("settings-copy.gradle");
+
+            // Rename settings.gradle to settings-duplicate.gradle
+            Files.move(originalPath, originalPath.resolveSibling("settings-duplicate.gradle"));
+            // Rename settings-copy.gradle to settings.gradle
+            Files.move(originalPathCopy, originalPathCopy.resolveSibling("settings.gradle"));
+
+            Path projectDirNewPath = pathNew.resolve(SM_MP_PROJECT_NAME_NEW);
+
+            // Rename the project directory to a new name, replacing it if it already exists
+            Files.move(projectDirPath, projectDirNewPath, StandardCopyOption.REPLACE_EXISTING);
+
+            // Prepare the environment with the new project path and name
+            prepareEnv(PROJECTS_PATH_NEW, SM_MP_PROJECT_NAME_NEW);
+
+        } catch (IOException e) {
+            System.err.println("Setup failed: " + e.getMessage());
+            e.printStackTrace();
+            Assertions.fail("Test setup failed due to an IOException: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Cleanup includes deleting the created project path.
+     */
+    @AfterAll
+    public static void cleanup() {
+        try {
+            closeProjectView();
+        } finally {
+            deleteDirectoryIfExists(PROJECTS_PATH_NEW);
```

Comment:
I have added this method in the finally block because it must execute even if an exception occurs. This ensures that the newly created directory is removed, leading to complete cleanup.